File: 2482-difference-between-ones-and-zeros-in-row-and-column/2482-difference-between-ones-and-zeros-in-row-and-column.cpp
class Solution {
public:
    vector<vector<int>> onesMinusZeros(vector<vector<int>>& grid) {
        int m=grid.size(); //rows
        int n=grid[0].size(); //columns
        vector<int> r(m,0);
        vector<int> c(n,0);
        for(int i=0; i<m; i++){
            for(int j=0; j<n; j++){
                if(grid[i][j]){
                    r[i]++; c[j]++;
                }
            }
        }

        for(int i=0; i<m; i++){
            for(int j=0; j<n; j++){
                grid[i][j]=2*(r[i]+c[j])-m-n;
            }
        }
        return grid;
        
    }
};

File: 2482-difference-between-ones-and-zeros-in-row-and-column/2482-difference-between-ones-and-zeros-in-row-and-column.py
class Solution:
    def onesMinusZeros(self, grid: List[List[int]]) -> List[List[int]]:
        l1=[]
        l2=[]
        for i in range(len(grid)):
            zero=0
            one=0
            for j in range(len(grid[i])):
                if(grid[i][j]==1):
                    one+=1
                else:
                    zero+=1
            l1.append([one,zero]) 

        for i in range(len(grid[0])):
            zero=0
            one=0
            for j in range(len(grid)):
                if(grid[j][i]==1):
                    one+=1
                else:
                    zero+=1
            l2.append([one,zero]) 
        # print(l1,l2)
        ans=grid.copy()
        # print(ans)
        for i in range(len(ans)):
            for j in range(len(ans[0])):
                ans[i][j] = l1[i][0]+l2[j][0]-l1[i][1]-l2[j][1]
        return ans
         